tools/tplg_parser/tplg_parser.c

```diff
@@ -555,7 +555,7 @@ int tplg_load_one_control(struct snd_soc_tplg_ctl_hdr **ctl, char **priv_data,
 	case SND_SOC_TPLG_DAPM_CTL_VOLSW:
 		/* load mixer type control */
 		rewind_size = sizeof(struct snd_soc_tplg_ctl_hdr);
-		if (fseek(file, rewind_size * -1, SEEK_CUR)) {
+		if (fseek(file, (long)rewind_size * -1, SEEK_CUR)) {
```

Comment:
are size_t and long always the same size?

Reply:
size_t is unsigned since negative size only makes sense in a black hole :)

Reply:
i never found a definitive answer on this from Google searches.

Reply:
seems like it can be `ULL`  on 64 bit and `U` on 32-bit, so 🤷